src/type/directives.ts

```diff
@@ -276,6 +281,54 @@ export const GraphQLOneOfDirective: GraphQLDirective = new GraphQLDirective({
   args: {},
 });
 
+/**
+ * Possible error handling actions.
+ */
+export const ErrorAction = {
+  PROPAGATE: 'PROPAGATE' as const,
+  NULL: 'NULL' as const,
+} as const;
+
+// eslint-disable-next-line @typescript-eslint/no-redeclare
+export type ErrorAction = (typeof ErrorAction)[keyof typeof ErrorAction];
+
+export const _ErrorAction = new GraphQLEnumType({
+  name: '_ErrorAction',
```

Comment:
This is `_ErrorAction` and not `__ErrorAction` because I get a validation error else. But ultimately, I'd vote for this to be `__ErrorAction`

Reply:
What's the validation error? Sorry can't run the code atm

Reply:
Complains that types starting with `__` are reserved for builtin types. Which makes sense but since it is not standard yet... But also, it might become standard...

Reply:
You need to add it to the introspection types I think:

https://github.com/graphql/graphql-js/blob/eb9b6c8a77193fc5ed3685d1e95d32b99c9df9d9/src/type/introspection.ts#L644-L655

That's where it should be defined too.

Reply:
It's an interesting one because it's _reserved_ but it's not actually part of _introspection_. So it should either become a well known (like `String`, `ID`, `Int`, `Boolean`, `Float`) or it should be reserved (`__`) which is currently "reserved for introspection". Since it can't be populated by a variable we don't need the type name in a document so it's not super important, but also feels weird to have an `__` type in a directive that we're expecting users to use heavily.

Reply:
>  it should be reserved (__) which is currently "reserved for introspection".

[TIL](https://spec.graphql.org/draft/#sel-IAFNZVDDABBBD0hT). I always thought of `__` as reserved irrespective of introspection. Is there a specific reason why this mentions `__` explicitely here?

Reply:
Now that the directive doesn't take arguments, the problem has disappeared 😄 . Should we explicitely reserve `@experimental_disableErrorPropagation` somewhere? Guessing it's the same issue with `@defer` and `@oneOf`. Is there a process to "reserve" directive names?

Reply:
No need to reserve it, it’s sufficiently complex IMO, extremely unlikely someone would implement this to do something else or to require arguments.

Reply:
Perfect 👍